lib/msf/core/opt_meterpreter_debug_logging.rb

```diff
@@ -0,0 +1,69 @@
+# -*- coding: binary -*-
+
+module Msf
+  ###
+  #
+  # Meterpreter Debug Logging option
+  #
+  ###
+  class OptMeterpreterDebugLogging < OptBase
+    def initialize(in_name, attrs = [], **kwargs)
+      super
+    end
+
+    def type
+      'meterpreterdebuglogging'
+    end
+
+    def validate_on_assignment?
+      true
+    end
+
+    def valid?(value, check_empty: true)
+      return false if !super(value, check_empty: check_empty)
+
+      begin
+        _parse_result = self.class.parse_logging_options(value)
+        true
+      rescue ::ArgumentError
+        false
+      end
+    end
+
+    ##
+    #
+    # Parses the given Meterpreter Debug Logging string
+    #
+    ##
+    def self.parse_logging_options(value)
+      result = {}
+      errors = []
+
+      return result if value.nil?
+
+      value = value.strip
+      # Match 'rpath:./file', 'rpath:/file', and drive letters e.g. 'rpath:C:/file'
+      rpath_regex = %r{^rpath:((\.?/\p{ASCII}+)|(\p{ASCII}:/\p{ASCII}+))}i
+
+      # Check if we log to rpath
+      if value.match?(rpath_regex)
+        # https://docs.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=cmd
+        max_length = 260
```

Comment:
This max length is for windows but correct me if I'm wrong this will apply to all OSs?

Reply:
Currently it will, yes. However as we are validating the `rpath:` value in this case upon typing in `set MeterpreterDebugLogging rpath:/xyz` (before we know the OS of the target), I'm not sure how/if we could handle this differently.
Correct me if I'm missing something obvious here 👍